zng/docs/spec.md

```diff
@@ -95,11 +109,15 @@ The stream of values is interleaved with control messages
 that provide type definitions and other metadata.  The type of
 a particular data value is specified by its "type identifier", or type ID,
 which is an integer representing either a "primitive type" or a
-"container type".
+"complex type".
 
 The ZNG type system comprises the standard set of primitive types like integers,
-floating point, strings, byte arrays, etc. as well as container types
-like records, arrays, and sets arranged from the primitive types.
+floating point, strings, byte arrays, etc. as well as complex types
+that are built from other types like records, arrays, and sets arranged
+from the primitive types.
+
+ZNG also includes first-class types, where a value can be of type "type".
+This is useful for introspection and grouping aggregations by type.
```

Comment:
```suggestion
This is useful for data exploration and grouping aggregations by type.
```

(maybe?)

Reply:
This reads funny to me when I just swap the terms.  I landed here
```
This is useful for grouping aggregations by type to do introspective data exploration.
```